Server/sprawl_calculator/wup_calculator/views.py

```diff
@@ -1,13 +1,68 @@
 from django.shortcuts import render
 from rest_framework.decorators import api_view
+from rest_framework.response import Response
+from rest_framework import status
+from datetime import datetime
+from .models import WUPCalculation
+from .data_definations import WUPCalculationResult, WUPCalculationRequestActivating, WUPCalculationRequestProcessing, WUPCalculationRequestCompleted, WUPCalculationRequestRejected, WUPCalculationRequestError
 # Create your views here.
 
-#Endpoint1_View
+
+""" Start a WUP index calculation processing request """
+
 @api_view(['PUT', 'GET'])
 def wup_index_generator(request):
     raise NotImplementedError
 
-#Endpoint2_View
+""" A endpoint to see results of a index calculation processing request """
+
 @api_view(['GET'])
-def index_calculation_status(request,processing_id):
-    raise NotImplementedError
\ No newline at end of file
+def index_calculation_status(request, processing_id):
+    try:
+        wup_calculation = WUPCalculation.objects.get(processing_id=processing_id)
+    except WUPCalculation.DoesNotExist:
+        return Response(status=status.HTTP_404_NOT_FOUND)
+
+    if wup_calculation.status == "Activating":
+        response_data = WUPCalculationRequestActivating(
+            status=wup_calculation.status,
+            processing_id=wup_calculation.processing_id,
+            created_at=wup_calculation.created_at,
+            updated_at=wup_calculation.updated_at
+        )
+    elif wup_calculation.status == "Processing":
+        response_data = WUPCalculationRequestProcessing(
+            status=wup_calculation.status,
+            processing_id=wup_calculation.processing_id,
+            created_at=wup_calculation.created_at,
+            updated_at=wup_calculation.updated_at
+        )
+    elif wup_calculation.status == "Completed":
+        response_data = WUPCalculationRequestCompleted(
+            status=wup_calculation.status,
+            processing_id=wup_calculation.processing_id,
+            result=WUPCalculationResult(
+                dis=wup_calculation.dis,
+                lup=wup_calculation.lup,
+                wup=wup_calculation.wup
+            ),
+            created_at=wup_calculation.created_at,
+            updated_at=wup_calculation.updated_at
+        )
+    elif wup_calculation.status == "Rejected":
+        response_data = WUPCalculationRequestRejected(
+            status=wup_calculation.status,
+            processing_id=wup_calculation.processing_id,
+            created_at=wup_calculation.created_at,
+            updated_at=wup_calculation.updated_at
+        )
+    else:
+        response_data = WUPCalculationRequestError(
+            status=wup_calculation.status,
+            processing_id=wup_calculation.processing_id,
+            created_at=wup_calculation.created_at,
+            updated_at=wup_calculation.updated_at
+        )
+
+    return Response(response_data.to_dict(), status=status.HTTP_200_OK)
```

Comment:
The Response shouold be a [JsonResponse](https://docs.djangoproject.com/en/dev/ref/request-response/#jsonresponse-objects) instead of just Response.